### PR TITLE
Add cocos new tests to travis CI

### PR DESCRIPTION
**Pull request by crazyhappygame**

### CODE REVIEW

**Comment by minggo** [comment]
Currently the CI is very slow and always doesn't trigger Mac/iOS building, so i don't think add more tasks is not a good idea.

**Comment by crazyhappygame** [comment]
OK. I will remove "cocos new" test for iOS/Mac

**Comment by drelaptop** [comment]
Travis error on macOS `URLError: <urlopen error [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:590)>`, the reason is SSL version in python lower than 1.01. 

Oh! I got it, upgrade SSL in python success

```sh
# fix error, URLError: <urlopen error [SSL: TLSV1_ALERT_PROTOCOL_VERSION]
function upgrade_openssl_for_osx()
{
    /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
    brew update
    brew upgrade openssl
    ln -s /usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib /usr/local/lib/
    ln -s /usr/local/opt/openssl/lib/libssl.1.0.0.dylib /usr/local/lib/
    ln -s /usr/local/Cellar/openssl/1.0.2n/bin/openssl /usr/local/bin/openssl
    echo "macOS SSL: `openssl version`"
    brew install python2 --with-brewed-openssl
    ln -s /usr/local/opt/python@2/bin/python2 /usr/local/bin/python
    echo "python SSL: `python -c "import ssl; print ssl.OPENSSL_VERSION"`"
}
```

have fixed this issue in #18683

**Comment by crazyhappygame** [comment]
@drelaptop thank you!
And what about https://travis-ci.org/cocos2d/cocos2d-x/jobs/347931577. Does "cocos new " build works for linux? In this PR it fails with following error:

```
[100%] Building CXX object CMakeFiles/TemplateCpp.dir/proj.linux/main.cpp.o
[100%] Building CXX object CMakeFiles/TemplateCpp.dir/Classes/AppDelegate.cpp.o
[100%] Building CXX object CMakeFiles/TemplateCpp.dir/Classes/HelloWorldScene.cpp.o
[100%] Linking CXX executable bin/TemplateCpp/TemplateCpp
/usr/bin/ld: ../cocos2d/external/bullet/prebuilt/linux/64-bit/libBulletDynamics.a(btConeTwistConstraint.o): unrecognized relocation (0x2a) in section `.text'
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
make[2]: *** [bin/TemplateCpp/TemplateCpp] Error 1
make[1]: *** [CMakeFiles/TemplateCpp.dir/all] Error 2
make: *** [all] Error 2
```

https://travis-ci.org/cocos2d/cocos2d-x/jobs/347931577
 It seems like libBullet is missing.

**Comment by drelaptop** [comment]
I tested `cocos new` on ubuntu16.04 in that cmake PR, it works well. so it not a lib issue, but a environment issue.  

I found your `linux_cocos_new_test ` didn't execute `install_linux_environment` function. see this code block:
```
# set up environment according os and target
function install_environement_for_pull_request()
{
    echo "Building pull request ..."

    if [ "$TRAVIS_OS_NAME" == "linux" ]; then
        if [ "$BUILD_TARGET" == "linux" ]; then
            install_linux_environment
        fi

        if [ "$BUILD_TARGET" == "android" ]; then
            install_android_environment
        fi
    fi
```
so you need to change the condition: ` if [ "$BUILD_TARGET" == "linux" ]; then`

As for the android didn't compile cpp files in travis, I think the reason it execute `cocos compile -p android` but cocos command line didn't consider cmake compile, see android build in AppVeyor it works well, it execute `./gradlew build`

File: tools/travis-scripts/run-script.sh [hunk 1]
```diff
@@ -41,24 +41,21 @@ function build_android()
...
-    cocos compile -p android
+    pushd $COCOS2DX_ROOT/tests/cpp-tests/proj.android
+    ./gradlew build
     popd
 
     # build lua-tests
-    pushd $COCOS2DX_ROOT/tests/lua-tests
-    cocos compile -p android
+    pushd $COCOS2DX_ROOT/tests/lua-tests/project/proj.android
+    ./gradlew build
```
Comment (drelaptop) [inline tools/travis-scripts/run-script.sh]: I found the android build always TimeOut, this task build the engine for every test project build. it build `cpp-tests` and `lua-tests`, spending 2 times the build engine time.

Can we divide this task into 2 travis task? I think TimeOut problem will be solved after divide.

**Comment by crazyhappygame** [comment]
@minggo @drelaptop Could you merge it?
Changes
1 "cocos new test" for android
2. "cocos new test" for linux
3. use "./gradlew assembleDebug" instead of cocos command

File: tools/travis-scripts/run-script.sh (hunk 1 above)
Comment (crazyhappygame) [reply tools/travis-scripts/run-script.sh]: I change it to "./gradlew assembleDebug". "cocos compile -p android" built debug only.
"./gradlew build" builds Debug and Release

**Comment by drelaptop** [comment]
only existed travis CI checks, so strange. where the `appveyor` CI checks？please check. or what's the reason to remove it

**Comment by crazyhappygame** [comment]
@drelaptop 
I do not understand why there was no appveyor... (maybe github/appveyor was unavailable for while). I did minor update and appveyor is back.
Travis Mac/iOS fails one again .... Could you check it?

**Comment by drelaptop** [comment]
got , I will do it tomorrow

**Comment by crazyhappygame** [comment]
Thank you

**Comment by drelaptop** [comment]
@slackmoehrle I was failed to solve this issue, we need your help for I know you are a so experienced guy when using of Homebrew.

the initial issue is `URLError: <urlopen error [SSL: TLSV1_ALERT_PROTOCOL_VERSION]`, to fix this, we need to upgrade the ssl in python2.7.

**Comment by drelaptop** [comment]
upgrade openssl version in python on mac platform by
```
brew unlink python
brew install python2
echo "python link: `ls -l /usr/local/bin/python`"
echo "python SSL: `python -c "import ssl; print ssl.OPENSSL_VERSION"`"
```

```
export PYTHON_BIN=/usr/bin/python
if [ "$TRAVIS_OS_NAME" == "osx" ]; then
     export PYTHON_BIN=/usr/local/bin/python
   fi
```
please fix Conflicting files, I think the travis will works well

**Comment by drelaptop** [comment]
Travis  CI passed except Android target, the android target build cpp-tests and lua-tests one by one, I suggest to divide this build task into 2 small task. So both can be built in the same time.

In the log, https://travis-ci.org/cocos2d/cocos2d-x/jobs/353192350, we can find build cpp-tests spend 30 minutes, both might spend 60 minutes, but the time limit 49 minutes

> BUILD SUCCESSFUL in 30m 8s
> 54 actionable tasks: 54 executed

**Comment by crazyhappygame** [comment]
OK. I split it. Android travis is very slow ... Just for your information cpp-test, lua-tests, js-tests are built in appveyor win32 as well.

**Comment by crazyhappygame** [comment]
@minggo @drelaptop Could you merge it?

**Comment by zcgit** [comment]
I got this error(<urlopen error [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:661)>) today in the version 3.17. and I've tried with the method like above but failed, so can we show a log to tell us how to download the zips by manual?, btw, my env is mac.

**Comment by minggo** [comment]
You can download v3.17 here: http://cocos2d-x.org/download. You can not download it from github, because it doesn't include submodules and 3rd party libraries.

**Comment by drelaptop** [comment]
> urlopen error [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:661)

issues related Stack overflow thread @zcgit 

https://stackoverflow.com/questions/44316292/ssl-sslerror-tlsv1-alert-protocol-version

It's a issue related with SSL module in python, you can select to upgrade your python with SSL. or download the 3rd party dependence in below link

https://github.com/cocos2d/cocos2d-x-3rd-party-libs-bin/releases

**Comment by zcgit** [comment]
> > urlopen error [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:661)
> 
> issues related Stack overflow thread @zcgit
> 
> https://stackoverflow.com/questions/44316292/ssl-sslerror-tlsv1-alert-protocol-version
> 
> It's a issue related with SSL module in python, you can select to upgrade your python with SSL. or download the 3rd party dependence in below link
> 
> https://github.com/cocos2d/cocos2d-x-3rd-party-libs-bin/releases

thx, I've downloaded it by manual, and it works well now.

**Pull request merged**